### PR TITLE
Validate that all options are successfully read from VS settings.

### PR DESCRIPTION
**Pull request by tmat**

### CODE REVIEW

File: src/VisualStudio/IntegrationTest/New.IntegrationTests/Options/GlobalOptionsTest.cs [hunk 1]
```diff
@@ -64,7 +64,10 @@ public async Task ValidateAllOptions()
             foreach (var language in option.IsPerLanguage ? allLanguages : noLanguages)
             {
                 var key = new OptionKey2(option, language);
-                var currentValue = globalOptions.GetOption<object?>(key);
+
+                // validate that reading the option directly from the persister without falling back to default value works:
+                AssertEx.AreEqual(true, vsSettingsPersister.TryFetch(key, out var currentValue),
+                    message: $"Option '{option.Definition.ConfigName}' failed to load from VS settings.");
```
Comment (genlu) [inline src/VisualStudio/IntegrationTest/New.IntegrationTests/Options/GlobalOptionsTest.cs:70]: Looks like this already caught something :)

```
Option 'SplitStringLiteralOptions_Enabled' failed to load from VS settings.
Expected:
True
Actual:
False
```

Comment (tmat) [reply src/VisualStudio/IntegrationTest/New.IntegrationTests/Options/GlobalOptionsTest.cs:70]: Haha, yeah, I think this change is needed first: https://github.com/dotnet/roslyn/pull/66421

Comment (tmat) [reply src/VisualStudio/IntegrationTest/New.IntegrationTests/Options/GlobalOptionsTest.cs:70]: Actually, seems like by many of the settings are not present in the registry in the default VS image. So we can't assume they are there. This test is not gonna work.

**Pull request closed without merging**